### PR TITLE
[OF-1195] feat!: Add XErrorCodes to WebService

### PR DESCRIPTION
**Pull request by Mag-JB**
connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/src/Services/WebService.cpp
Comment (MAG-SamBirley) [inline Library/src/Services/WebService.cpp:116]: Worth including a log message here to help us catch error codes that we don't yet track in CSP?

**Review by MAG-SamBirley — CHANGES_REQUESTED** [review]

File: Library/include/CSP/Services/WebService.h
Comment (MAG-AdamThorn) [inline Library/include/CSP/Services/WebService.h:149]: Does this still need to be virtual?

File: Library/src/Services/WebService.cpp
Comment (MAG-SamBirley) [reply Library/src/Services/WebService.cpp]: Saw the new error message. Super useful! Thanks :)

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [reply Library/include/CSP/Services/WebService.h]: No, and it should also be made private.

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [inline Library/include/CSP/Services/WebService.h:161]: This map should be moved to WebService.cpp in an anonymous namespace, changes to a `std::map`, and be made static. We don't want to re-create this every time a new ResultBase instance is made. We should also only be mapping the quota errors and any error codes we already handle as that is all we need to cover in this ticket.

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [inline Library/include/CSP/Services/WebService.h:63]: Can we get a better name for this now that it contains all error codes?

**Review by MAG-MichaelK — CHANGES_REQUESTED** [review]

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [reply Library/include/CSP/Services/WebService.h:63]: Something like `EFailureReason`

**Review by MAG-SamBirley — APPROVED** [review]
Happy with the changes made to address my requests, so approved from that point of view. I would like to also see the feedback from @MAG-MichaelK addressed too though!

**Review by MAG-MichaelK — APPROVED** [review]

**Comment by Mag-JB** [comment]
I will test this before merging!

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [inline Library/include/CSP/Services/WebService.h:148]: Can we also make this private as it doesn't need to be accessed by any derived types?

**Review by MAG-MichaelK — CHANGES_REQUESTED** [review]

**Review by MAG-SamBirley — DISMISSED** [review]

File: Library/include/CSP/Services/WebService.h
Comment (MAG-MichaelK) [reply Library/include/CSP/Services/WebService.h:148]: Just waiting on this to be done then I can approve this PR

Comment (Mag-JB) [reply Library/include/CSP/Services/WebService.h:148]: oops didn't see this one sneak in after the approval

**Review by MAG-MichaelK — APPROVED** [review]

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**
